src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtins/gtir_builtin_field_operator.py

```diff
@@ -0,0 +1,135 @@
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+from typing import Callable
+
+import dace
+
+from gt4py.next.common import Dimension
+from gt4py.next.iterator import ir as itir
+from gt4py.next.iterator.ir_utils import common_pattern_matcher as cpm
+from gt4py.next.program_processors.runners.dace_fieldview.gtir_builtins.gtir_builtin_translator import (
+    GTIRBuiltinTranslator,
+)
+from gt4py.next.program_processors.runners.dace_fieldview.gtir_dataflow_builder import (
+    GTIRDataflowBuilder,
+)
+from gt4py.next.program_processors.runners.dace_fieldview.gtir_tasklet_codegen import (
+    GTIRTaskletCodegen,
+)
+from gt4py.next.type_system import type_specifications as ts
+
+
+class GTIRBuiltinAsFieldOp(GTIRBuiltinTranslator):
+    """Generates the dataflow subgraph for the `as_field_op` builtin function."""
+
+    stencil_expr: itir.Lambda
+    stencil_args: list[Callable]
+    field_domain: dict[Dimension, tuple[str, str]]
+    field_type: ts.FieldType
+
+    def __init__(
+        self,
+        dataflow_builder: GTIRDataflowBuilder,
+        state: dace.SDFGState,
+        node: itir.FunCall,
+        stencil_args: list[Callable],
+    ):
+        super().__init__(state, dataflow_builder.sdfg)
+
+        assert cpm.is_call_to(node.fun, "as_fieldop")
+        assert len(node.fun.args) == 2
+        stencil_expr, domain_expr = node.fun.args
+        # expect stencil (represented as a lambda function) as first argument
+        assert isinstance(stencil_expr, itir.Lambda)
+        # the domain of the field operator is passed as second argument
+        assert isinstance(domain_expr, itir.FunCall)
+
+        # visit field domain
+        domain = dataflow_builder.visit_domain(domain_expr)
+
+        # add local storage to compute the field operator over the given domain
+        # TODO: use type inference to determine the result type
+        node_type = ts.ScalarType(kind=ts.ScalarKind.FLOAT64)
+
+        self.field_domain = {dim: (lb, ub) for dim, lb, ub in domain}
+        self.field_type = ts.FieldType([dim for dim, _, _ in domain], node_type)
+        self.stencil_expr = stencil_expr
+        self.stencil_args = stencil_args
+
+    def build(self) -> list[tuple[dace.nodes.Node, ts.FieldType | ts.ScalarType]]:
+        # generate a tasklet node implementing the stencil function and represent
+        # the field operator as a mapped tasklet, which will range over the field domain
+        output_connector = "__out"
+        tlet_code = "{var} = {code}".format(
+            var=output_connector, code=GTIRTaskletCodegen().visit(self.stencil_expr.expr)
+        )
+
+        # allocate local temporary storage for the result field
+        field_shape = [
```

Comment:
Note to myself: does the data have to flow over a "local storage"

Reply:
For the final write to an external field, this internal node is redundant, and dace simplify pass will remove it. On the other hand, it is needed when we write back to a field also used as input (in-out field parameters): in this case, the dace simplify pass will correctly keep it. I have added a testcase `test_gtir_update` to cover this case.